File: Ontology/Entity.cs
namespace Ontology
{
    public abstract class Entity:
        Axiom,
        IEntity
    {
        protected readonly string _localName;

        protected Entity(
            IOntology ontology,
            string    localName
            ) : base(ontology)
        {
            _localName = localName;
        }

        string IEntity.Iri => _ontology.Iri + '.' + _localName;

        string IEntity.LocalName => _localName;

        public override string ToString() => _localName;
    }
}

File: Web/ClientApp/src/app/Ontology/Entity.ts
import { Axiom } from "./Axiom";
import { IEntity } from "./IEntity";
import { IOntology } from "./IOntology";

export class Entity
    extends Axiom
    implements IEntity
{
    protected constructor(
        ontology        : IOntology,
        public LocalName: string
        )
    {
        super(ontology);
    }

    get PrefixIri(): string
    {
        return `${this.Ontology.Iri}.`;
    }

    get Iri(): string
    {
        return `${this.PrefixIri}${this.LocalName}`;
    }
}
